### PR TITLE
Loadgen startup fix

### PR DESCRIPTION
**Pull request by marta-lokhova**
Resolves https://github.com/stellar/stellar-core/issues/4071

### CODE REVIEW

File: src/simulation/LoadGenerator.cpp [hunk 1]
```diff
@@ -421,7 +376,92 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
...
+GeneratedLoadConfig::areTxsRemaining() const
+{
+    return (isCreate() && nAccounts != 0) || (!isCreate() && nTxs != 0);
+}
+
+void
+LoadGenerator::start(GeneratedLoadConfig& cfg)
+{
+    // Setup config for soroban modes
+    if (!mStarted && cfg.isSoroban())
```
Comment (SirTyson) [inline src/simulation/LoadGenerator.cpp:420]: `start` should be a no-op except on the first invocation, so why not add `if (mStarted) return;` at the beginning and get rid of the checks in the individual ifs. Currently the only functions that get called multiple times in `start` are `createRootAccount` and `updateMinBalance`. If we need to call those periodically they probably don't belong in `start` (but I don't think we do).

File: src/simulation/LoadGenerator.cpp [hunk 2]
```diff
@@ -458,43 +498,16 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
...
-{
-    return (isCreate() && nAccounts == 0) || (isLoad() && nTxs == 0) ||
-           (isSorobanSetup() && nInstances == 0);
-}
-
-bool
-GeneratedLoadConfig::areTxsRemaining() const
-{
-    return (isCreate() && nAccounts != 0) || (!isCreate() && nTxs != 0);
+    createRootAccount();
```
Comment (SirTyson) [inline src/simulation/LoadGenerator.cpp:509]: Now that we have a single start function, we should change `if (!mRoot)` to `releaseAssert(!mRoot)` in `createRootAccount()`.

File: src/simulation/LoadGenerator.cpp [hunk 3]
```diff
@@ -458,43 +498,16 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
...
     mStarted = true;
 
-    if (mApp.getState() == Application::APP_SYNCED_STATE)
-    {
-        mLoadTimer->expires_from_now(std::chrono::milliseconds(STEP_MSECS));
-        mLoadTimer->async_wait([this, cfg]() { this->generateLoad(cfg); },
-                               &VirtualTimer::onFailureNoop);
-    }
-    else
+    if (!mStartTime)
```
Comment (SirTyson) [inline src/simulation/LoadGenerator.cpp:503]: I think we should change this to `releaseAssert(!mStartTime)` (assuming you implement my other comment where we exit early in `start()`).

**Review by SirTyson — CHANGES_REQUESTED** [review]
Looks good! I think we can clean up the start function a little more though.

**Comment by marta-lokhova** [comment]
@SirTyson added your suggestions, so this should be ready now

File: src/simulation/LoadGenerator.cpp [hunk 4]
```diff
@@ -345,6 +343,67 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
         }
     }
 
+    if (cfg.mode != LoadGenMode::CREATE)
+    {
+        {
```
Comment (SirTyson) [inline src/simulation/LoadGenerator.cpp:348]: Nit: Extra brackets

**Review by SirTyson — APPROVED** [review]
LGTM, one small nit.

File: src/simulation/LoadGenerator.cpp (hunk 4 above)
Comment (marta-lokhova) [reply src/simulation/LoadGenerator.cpp:350]: oops, fixed

**Comment by sisuresh** [comment]
r+ fce809d6f1121539cb29b39580b05f55d56c21b2

**Pull request merged**